lib/pbench/cli/agent/commands/tools/base.py

```diff
@@ -0,0 +1,28 @@
+from pbench.agent.base import BaseCommand
+
+
+class ToolCommand(BaseCommand):
+    """Common Tool Class"""
+
+    def __init__(self, context):
+        super(ToolCommand, self).__init__(context)
+
+    def remote(self, path):
+        """List all remotes in a given path"""
+        return [p.name for p in path.iterdir() if p.name != "__trigger__"]
+
+    def tools(self, path):
+        """List all tools in a given path"""
+        _tools = []
+        for p in path.glob("*"):
```

Comment:
Curious, is there a functional difference between `iterdir()` and `glob("*")`?

And it looks like a list comprehension would work here, worth it (other two methods use them)?

```
_tools = [p.name for p in path.iterdir() if p.name != "__label__" && p.suffix != ".__noinstall__"]
```

Reply:
fixed 

glob is better when you need to make use of the filtering feature and the filter is simple and string-based, as it simplifies the work. Sure, hand-writing simple matches (filtering iterdir via if path.endswith('.txt.'): instead of glob('*.txt')) might be more efficient than the regex based pattern matching glob hides, but it's generally not worth the trouble of reinventing the wheel given that disk I/O is orders of magnitude slower.